package/MDAnalysis/coordinates/array.py

```diff
@@ -0,0 +1,240 @@
+        :Arguments:
+            *coordinate_array*
+               :class:`~numpy.ndarray object
+            *format*
+               the order/shape of the return data array, corresponding
+               to (a)tom, (f)rame, (c)oordinates all six combinations
+               of 'a', 'f', 'c' are allowed ie "fac" - return array
+               where the shape is (frame, number of atoms,
+               coordinates)
+        """
+        self.set_array(coordinate_array, format)
+        self.n_frames = coordinate_array.shape[self.format.find('f')]
+        self.n_atoms = coordinate_array.shape[self.format.find('a')]
+
+        kwargs.pop("n_atoms", None)
+        self.ts = self._Timestep(self.n_atoms, **kwargs)
+        # self.ts.frame = -1
+        self._read_next_timestep()
+
+    def set_array(self, coordinate_array, format='afc'):
+        """
+        Set underlying array in desired column format.
+
+        :Arguments:
+            *coordinate_array*
+               :class:`~numpy.ndarray object
+            *format*
+               the order/shape of the return data array, corresponding
+               to (a)tom, (f)rame, (c)oordinates all six combinations
+               of 'a', 'f', 'c' are allowed ie "fac" - return array
+               where the shape is (frame, number of atoms,
+               coordinates)
+
+        """
+        self.coordinate_array = coordinate_array
+        self.format = format
+
+    def get_array(self, format='afc'):
+        """
+        Return underlying array in desired column format.
+        This methods has overlapping functionality with the
+        timeseries method, but is slightly faster in cases
+        where no selection or filtering is required. Another
+        difference is that get_array always returns a view of
+        the original array, while timeseries will return a copy.
+
+        :Arguments:
+            *format*
+               the order/shape of the return data array, corresponding
+               to (a)tom, (f)rame, (c)oordinates all six combinations
+               of 'a', 'f', 'c' are allowed ie "fac" - return array
+               where the shape is (frame, number of atoms,
+               coordinates)
+        """
+        array = self.coordinate_array
+        if format==self.format:
+            pass
+        elif format[0] == self.format[0]:
+            array = np.swapaxes(array, 1, 2)
+        elif format[1] == self.format[1]:
+            array = np.swapaxes(array, 0, 2)
+        elif format[2] == self.format[2]:
+            array = np.swapaxes(array, 0, 1)
+        elif self.format[1] == format[0]:
+            array = np.swapaxes(array, 1, 0)
+            array = np.swapaxes(array, 1, 2)
+        elif self.format[2] == format[0]:
+            array = np.swapaxes(array, 2, 0)
+            array = np.swapaxes(array, 1, 2)
+        return array
+
+
+    def _reopen(self):
+        """Reset iteration to first frame"""
+        self.ts.frame = -1
+
+    def __iter__(self):
```

Comment:
This isn't required as you're already checking `ts.frame` in  `_read_next_timestep`, so the `base.__iter__` will work here